core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -2173,6 +2225,14 @@ object UnifiedLog extends Logging {
     }
   }
 
+  private[log] def localRetentionMs(config: LogConfig): Long = {
+    if (config.remoteLogConfig.remoteStorageEnable) config.remoteLogConfig.localRetentionMs else config.retentionMs
+  }
+
+  private[log] def localRetentionSize(config: LogConfig): Long = {
+    if (config.remoteLogConfig.remoteStorageEnable) config.remoteLogConfig.localRetentionBytes else config.retentionSize
+  }
+
```

Comment:
You can instead use similar methods already present in LogConfig.

see LogConfig.localRetentionBytes() and LogConfig.localRetentionMs()

(you will probably have to modify them to add new case of `if (config.remoteLogConfig.remoteStorageEnable)`